### ISSUE TITLE
Two issues caused a black screen/closed connection and how I fixed them on a fresh XRDP Install on Ubuntu 18.04

### ISSUE THREAD

**Issue by hpka** [neutrinolabs/xrdp#1294]
In short, I installed XRDP as follows on a fresh Ubuntu Server 18.04 LTS VM (followed several guides but this is the short version):
`sudo apt update && sudo apt upgrade -y && sudo apt install xrdp xfce4 xrdpxorg`

Problem: After connecting via mstsc and logging into XRDP, the connection screen would go black and close.

These errors were found by running (got the idea [here](https://github.com/neutrinolabs/xrdp/issues/1128)):

`sudo journalctl -S -3m`

... and seeing the helpfully highlighted red text.

First issue: 

`Feb 15 17:08:42 XXX xrdp[2413]: (2413)(140402390107968)[ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied`

Solution: 

`sudo chown root:users /etc/xrdp/key.pem`

Second issue:

`Feb 15 17:08:35 towerubuntu xrdp-sesman[2353]: (2353)(140649974506496)[INFO ] /usr/lib/xorg/Xorg :16 -auth .Xauthority -config xrdp/xorg.conf -noreset -nolisten tcp -logfile .xorgxrdp.%s
Feb 15 17:08:35 towerubuntu xrdp-sesman[2351]: (2351)(140649974506496)[CORE ] waiting for window manager (pid 2352) to exit
Feb 15 17:08:35 towerubuntu xrdp-sesman[2351]: (2351)(140649974506496)[CORE ] window manager (pid 2352) did exit, cleaning up session`

Solution (sorry, can't find the link now, but full credit to the person who gave me the idea): 

`nano .xsession`

It read:

`"xfce4-session"
`

Changed too:

`xfce4-session`

(without the quotation marks).

After that, xrdp worked fine post login.

Why am I posting this? Honestly just in case someone find's it helpful.

**Comment by davewthompson** [comment]
Just a note - should it be "xorgxrdp" rather than "xrdpxorg" in the instructions?

**Comment by metalefty** [comment]
Thank you very much for the information.

> Why am I posting this? Honestly just in case someone find's it helpful.

Just a note for someone, closing.

**Comment by gutors** [comment]
the changing of the cinnamon-session to xfce4-session worked for me.

Thank u!

**Comment by S-LR** [comment]
@gutors can you please explain how to change this ? would be very helpful

thanks alot and have a nice day
slr

**Comment by rnovicky** [comment]
After a while I've just created the missing .xsession file with `xfce4-session` and problem solved. Thanks!

**Comment by jslay88** [comment]
yup, some how i wiped out my .xsession file and this helped me track the issue down. Thanks!

**Comment by gahlberg** [comment]
can you share and post your .xsession file? for an example?

**Comment by rnovicky** [comment]
@gahlberg My .xsession file contains just one line:
`xfce4-session`

**Comment by tedekeroth** [comment]
> 
> First issue:
> 
> `Feb 15 17:08:42 XXX xrdp[2413]: (2413)(140402390107968)[ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied`
> 
> Solution:
> 
> `sudo chown root:users /etc/xrdp/key.pem`
> 


Thanks fot that post. However, it doesnt work for me. I tried running that command, and the command was executed, but I still get the "Permission denied".

So I proceeded with doing the same for /etc/ssl/certs/ssl-cert-snakeoil.pem (the key.pem seesm to be  linked to that one), but it made no difference.

Then I did "sudo chmod 777 key.pem", but no difference.


LOG:

```

[20210604-10:30:11] [DEBUG] xrdp_wm_log_msg: some problem
[20210604-10:30:11] [DEBUG] xrdp_mm_module_cleanup
[20210604-10:30:11] [DEBUG] Closed socket 16 (AF_INET6 ::1 port 58248)
[20210604-10:30:11] [DEBUG] Closed socket 12 (AF_INET6 ::ffff:192.168.10.208 port 3389)
[20210604-10:30:11] [DEBUG] xrdp_mm_module_cleanup
[20210604-10:30:44] [DEBUG] Closed socket 11 (AF_INET6 :: port 3389)
[20210604-10:30:44] [DEBUG] Testing if xrdp can listen on 0.0.0.0 port 3389.
[20210604-10:30:44] [DEBUG] Closed socket 7 (AF_INET6 :: port 3389)
[20210604-10:30:46] [INFO ] starting xrdp with pid 17422
[20210604-10:30:46] [INFO ] listening to port 3389 on 0.0.0.0
[20210604-10:30:53] [INFO ] Socket 12: AF_INET6 connection received from ::ffff:192.168.10.109 port 59748
[20210604-10:30:53] [DEBUG] Closed socket 12 (AF_INET6 ::ffff:192.168.10.208 port 3389)
[20210604-10:30:53] [DEBUG] Closed socket 11 (AF_INET6 :: port 3389)
[20210604-10:30:53] [INFO ] Using default X.509 certificate: /etc/xrdp/cert.pem
[20210604-10:30:53] [INFO ] Using default X.509 key file: /etc/xrdp/key.pem
[20210604-10:30:53] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20210604-10:30:53] [DEBUG] TLSv1.2 enabled
[20210604-10:30:53] [DEBUG] TLSv1.1 enabled
[20210604-10:30:53] [DEBUG] TLSv1 enabled
[20210604-10:30:53] [DEBUG] Security layer: requested 11, selected 0
[20210604-10:30:53] [DEBUG] Closed socket 12 (AF_INET6 ::ffff:192.168.10.208 port 3389)
[20210604-10:30:53] [INFO ] Socket 12: AF_INET6 connection received from ::ffff:192.168.10.109 port 59749
[20210604-10:30:53] [DEBUG] Closed socket 11 (AF_INET6 :: port 3389)
[20210604-10:30:53] [DEBUG] Closed socket 12 (AF_INET6 ::ffff:192.168.10.208 port 3389)
[20210604-10:30:53] [INFO ] Using default X.509 certificate: /etc/xrdp/cert.pem
[20210604-10:30:53] [INFO ] Using default X.509 key file: /etc/xrdp/key.pem
[20210604-10:30:53] [ERROR] Cannot read private key file /etc/xrdp/key.pem: Permission denied
[20210604-10:30:53] [DEBUG] TLSv1.2 enabled
[20210604-10:30:53] [DEBUG] TLSv1.1 enabled
[20210604-10:30:53] [DEBUG] TLSv1 enabled
[20210604-10:30:53] [DEBUG] Security layer: requested 0, selected 0
[20210604-10:30:53] [INFO ] connected client computer name: TED-DESKTOP
[20210604-10:30:53] [INFO ] Non-TLS connection established from ::ffff:192.168.10.109 port 59749: encrypted with standard RDP security
[20210604-10:30:53] [DEBUG] xrdp_0000441c_wm_login_mode_event_00000001
[20210604-10:30:53] [INFO ] Loading keymap file /etc/xrdp/km-0000041d.ini
[20210604-10:30:53] [WARN ] local keymap file for 0x0000041d found and doesn't match built in keymap, using local keymap file
[20210604-10:30:58] [DEBUG] xrdp_wm_log_msg: connecting to sesman ip 127.0.0.1 port 3350
[20210604-10:30:58] [INFO ] xrdp_wm_log_msg: sesman connect ok
[20210604-10:30:58] [DEBUG] xrdp_wm_log_msg: sending login info to session manager, please wait...
[20210604-10:30:58] [DEBUG] return value from xrdp_mm_connect 0
[20210604-10:30:59] [INFO ] xrdp_wm_log_msg: login successful for display 10
[20210604-10:30:59] [DEBUG] xrdp_wm_log_msg: started connecting
[20210604-10:30:59] [INFO ] lib_mod_log_peer: xrdp_pid=17436 connected to X11rdp_pid=17448 X11rdp_uid=1000 X11rdp_gid=1000 client_ip=::f$
[20210604-10:30:59] [DEBUG] xrdp_wm_log_msg: connected ok
[20210604-10:30:59] [DEBUG] xrdp_mm_connect_chansrv: chansrv connect successful
[20210604-10:30:59] [DEBUG] Closed socket 16 (AF_INET6 ::1 port 58258)
[20210604-10:30:59] [INFO ] The following channel is allowed: rdpdr (0)
[20210604-10:30:59] [INFO ] The following channel is allowed: rdpsnd (1)
[20210604-10:30:59] [INFO ] The following channel is allowed: cliprdr (2)
[20210604-10:30:59] [INFO ] The following channel is allowed: drdynvc (3)
[20210604-10:30:59] [DEBUG] The allow channel list now initialized for this session
[20210604-10:31:00] [DEBUG] Closed socket 18 (AF_UNIX)
[20210604-10:31:00] [DEBUG] Closed socket 12 (AF_INET6 ::ffff:192.168.10.208 port 3389)
[20210604-10:31:00] [DEBUG] xrdp_mm_module_cleanup
[20210604-10:31:00] [DEBUG] Closed socket 17 (AF_UNIX)

```

**Comment by CrazyCoder76** [comment]
Thanks for your help.
